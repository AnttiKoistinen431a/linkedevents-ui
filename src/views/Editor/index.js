
// styles
import '!style!css!sass!./index.scss'
import 'style!vendor/stylesheets/typeahead.css'

// js
import typeahead from 'typeahead.js'

// react-specific
import Formsy from 'formsy-react'
import React from 'react'
import {connect} from 'react-redux'
import {sendData, clearData, clearFlashMsg, fetchKeywordSets} from 'src/actions/editor.js'

import {FormattedMessage} from 'react-intl'

import Snackbar from 'material-ui/lib/snackbar';

import {
    Input,
    Textarea,
    Row
} from 'formsy-react-components'

import { RaisedButton, FlatButton } from 'material-ui'

// our components
import FF from 'src/formfields.js'
import Typeahead from 'src/typeahead.js'
import API from 'src/api.js'

// === constants ===

// the backup doesn't support non-language links, so we use hardcoded
// 'fi' instead for the link language
var EXT_LINK_NO_LANGUAGE = 'fi'

import FormFields from 'src/components/FormFields'

// === code ===
//

var EditEventForm = React.createClass({

    getInitialState() {
        return {
            canSubmit: false,
            layout: 'horizontal',
            validatePristine: false,
            disabled: false,
            location_id: '',
        }
    },

    componentDidUpdate() {
        // @trackActionChanges()

        // set up typeahead

        var taOptions = {
            hint: true,
            highlight: true,
            minLength: 1
        };

        var taDatasets = {
            limit: 25,
            display: 'value',
            templates: {
                suggestion(model) {
                   return '<div><class="place_name">' +
                   model.value +
                   '</span><br><span class="place_address">' +
                   model.street_address +
                   '</span></div>';
                },
                empty(d) {
                    return '<p class="repo-name">No Matches</p>';
                }
            },
            source: Typeahead.bloodhoundInstance.ttAdapter()
        };

        var taSelectHandler = (function(evt, item) {
            this.setState({
                location_id: item.id
            });
            return this.refs.__location_search_field.value = item.value;
        }
        ).bind(this);

        var taEm = $('#__location_search_field');
        taEm.typeahead(taOptions, taDatasets);
        return taEm.on('typeahead:selected', taSelectHandler);
    },

    enableButton() {
        return this.setState({
            canSubmit: true
        });
    },

    disableButton() {
        return this.setState({
            canSubmit: false
        });
    },

    clearForm() {
        this.props.dispatch(clearData())
    },

    willReceiveProps() {
        this.forceUpdate()
    },

    componentWillMount() {
        this.props.dispatch(fetchKeywordSets())
    },

    goToPreview(event) {
        console.log(event)
    },

    handleSubmit(event) {
        let doUpdate = this.props.action === 'update'
        this.props.dispatch(sendData(this.props.editor.values, this.props.user, doUpdate))
    },

    render() {
        var sharedProps = {
            layout: this.state.layout,
            validatePristine: this.state.validatePristine,
            disabled: this.state.disabled
        };

        // TODO: move to scss
        let paddingStyle = { paddingBottom: "1em" };
        let greenStyle = {color: 'green'};
        let blueStyle = {color: 'blue'};

        let buttonStyle = {
            height: '72px',
            margin: '0 10px'
        }

        let flashMsg = 'No message'
        if(this.props.editor.flashMsg) {
            flashMsg = (<FormattedMessage id={this.props.editor.flashMsg.msg} />)
        }

        let headerTextId = (this.props.action === 'update') ? 'edit-event' : 'create-event'

        return (
            <div>
                <div className="container header">
                    <h1>
                        <FormattedMessage id={headerTextId}/>
                    </h1>
                    <span className="controls">
                        <RaisedButton onClick={this.clearForm} primary={true} className="pull-right" label={<span><FormattedMessage id="clear-form"/><i className="material-icons">&#xE14C;</i></span>}/>
                    </span>
                </div>
                <Formsy.Form className="form-horizontal"
                             onSubmit={this.handleSubmit}
                             onValid={this.enableButton}
                             onInvalid={this.disableButton}
                             ref="editForm"
                             >
                    <div className="container">
                        <FormFields action={this.props.action} editor={this.props.editor} />
                    </div>

                    <div className="editor-action-buttons">
                        <div className="container">
                            <div className="row">
                                <div className="spread-right">
                                    <RaisedButton
                                        style={buttonStyle}
                                        label="Tallenna vedokseksi"
                                        onClick={ (e) => this.goToPreview(e) }
                                    />
                                    <RaisedButton
                                        style={buttonStyle}
                                        label="Siirry esikatseluun"
                                        primary={true}
                                        onClick={ (e) => this.goToPreview(e) }
                                    />
                                    <FlatButton
                                        style={buttonStyle}
                                        label="Julkaise tapahtuma"
                                        onClick={ (e) => this.handleSubmit(e) }
                                    />
                                </div>
                            </div>
                        </div>
                    </div>
                </Formsy.Form>
                <Snackbar
                  open={(!!this.props.editor.flashMsg)}
                  message={flashMsg}
                  bodyStyle={{'backgroundColor': 'rgb(0,108,188)'}}
                  autoHideDuration={6000}
                  onRequestClose={(e) => this.props.dispatch(clearFlashMsg())}
                />
            </div>
        )
    }
});

EditEventForm = connect((state) => ({
    editor: state.editor,
    user: state.user
}))(EditEventForm)

var EditorPage = React.createClass({

    getInitialState() {
        return {
            data: {},
            layout: 'horizontal',
            apiErrorMsg: ''
        }
    },

    getLocation(d) {
        var ret = null;
        if ('location' in d && d['location']) {
            if ('@id' in d['location'] && d['location']['@id']) {
                $.ajax({
                    type: 'GET',
                    url: d['location']['@id'],
                    dataType: 'json',
                    complete(response) {
                        return ret = $.parseJSON(response.responseText);
                    },
                    async: false
                });
            }
        }
        return ret;
    },

    render() {
        return (
            <div className="editor-page">
                <EditEventForm
                    eventId={this.props.params.eventId}
                    switchToPreview={this.switchToPreview}
                    updateData={this.updateData}
                    ref="formContainer"
                    action={this.props.params.action}
                    editor={this.props.editor}
                />
            </div>
        )
    }
});

export default connect((state) => ({
    editor: state.editor,
    user: state.user
}))(EditorPage)
