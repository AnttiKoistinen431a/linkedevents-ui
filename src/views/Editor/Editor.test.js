import configureStore from 'redux-mock-store'
import React from 'react'
import thunk from 'redux-thunk'
import {shallow, mount} from 'enzyme'
import renderer from 'react-test-renderer'

import testReduxIntWrapper from '__mocks__/testReduxIntWrapper'
import Editor from './index'
import {mockUser, mockEditorNewEvent, mockEditorExistingEvent} from '../../../__mocks__/mockData';

const mockStore = configureStore([thunk])
const initialStoreNewEvent = {
    subEvents: {
        isFetching: false,
        fetchComplete: false,
        items: [],
        error: null,
    },
    user: mockUser,
    editor: mockEditorNewEvent,
}
const initialStoreExistingEvent = {
    subEvents: {
        isFetching: false,
        fetchComplete: false,
        items: [],
        error: null,
    },
    user: mockUser,
    editor: mockEditorExistingEvent,
}

/*
    jest snapshot has node as null by default,
    so Component that relies on elementRef will fail tests,
    we mock the node here
*/
const mockNodeForRefInEditor = (element) => {
    return {style: {}};
};

describe('Editor Snapshot', () => {
    let store;
    const options = {
        createNodeMock: mockNodeForRefInEditor,
    };

    it('should render view correctly when new event', () => {
        store = mockStore(initialStoreNewEvent)
        const componentProps = {
            match: {
                params: {
                    action: 'create',
                    eventId: 'new?_k=dn954b',
                },
            },
        } // Props which are added to component
        const componentToTest = <Editor {...componentProps} />
        const wrapper = renderer.create(
            testReduxIntWrapper(store, componentToTest),
            options
        )
        expect(wrapper).toMatchSnapshot()
    })

    it('should render view correctly when editing existing event', () => {
        store = mockStore(initialStoreExistingEvent)
        const componentProps = {
            match: {
                params: {
                    action: 'update',
                    eventId: 'helsinki:afqxukccli',
                },
            },
        } // Props which are added to component
        const componentToTest = <Editor {...componentProps} />
        const wrapper = renderer.create(
            testReduxIntWrapper(store, componentToTest),
            options
        )
        expect(wrapper).toMatchSnapshot()
    })
})
