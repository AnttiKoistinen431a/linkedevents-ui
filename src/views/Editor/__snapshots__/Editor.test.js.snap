// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Editor Snapshot should render view correctly when editing existing event 1`] = `
<IntlProvider
  locale="fi"
  messages={
    Object {
      "add-more": "Lisää kerrat",
      "authorization-required": "Kirjaudu sisään tapahtuman lähettämiseksi.",
      "cancel": "Kumoa",
      "cancel-event": "Peruuta tapahtuma",
      "cancelled": "Peruutettu",
      "choose-image": "Lisää tapahtumalle kuva tai valitse kuvapankista",
      "clear-form": "Tyhjennä kentät",
      "confirm": "Suorita",
      "confirm-action-header": "Varmista toiminto",
      "confirm-cancel": "Varmista tapahtuman peruminen",
      "confirm-delete": "Varmista tapahtuman poistaminen",
      "confirm-image-delete": "Varmista kuvan poistaminen",
      "copy-event-to-draft": "Kopioi uuden tapahtuman pohjaksi",
      "copy-to-clipboard": "Kopioi leikepöydälle",
      "create-event": "Lisää uusi tapahtuma",
      "date": "Päivämäärä",
      "delete": "Poista",
      "delete-event": "Poista tapahtuma",
      "draft": "Luonnos",
      "edit-event": "Muokkaa tapahtumaa",
      "editor-delete-subevents-warning": "Poistaessasi tämän tapahtuman myös seuraavat alitapahtumat poistetaan:",
      "editor-delete-warning": "VAROITUS: Tämä toiminto poistaa tapahtuman lopullisesti. Voit tarvittaessa myös perua tapahtuman tai lykätä sitä.",
      "editor-sentinel-alert": "Voit katsella lomaketta, mutta sinulla ei ole oikeuksia julkaista tai muokata tapahtumia. Et ole kirjautunut sisään tai kirjautumisesi on vanhentunut.",
      "editor-tip-event-languages": "Kielet, joita tapahtumassa käytetään.",
      "editor-tip-hel-main-category": "Valitse vähintään yksi pääkategoria.",
      "editor-tip-hel-target-group": "Jos tapahtumalla ei ole erityistä kohderyhmää, älä valitse mitään.",
      "editor-tip-keywords": "Liitä tapahtumaan vähintään yksi asiasana, joka kuvaa tapahtuman teemaa. Aloita kirjoittamaan asiasanaa ja valitse lisättävä asiasana alle ilmestyvästä listasta.",
      "editor-tip-location": "Aloita kirjoittamaan kenttään tapahtumapaikan nimen alkua ja valitse oikea paikka alle ilmestyvästä listasta.",
      "editor-tip-location-extra": "Jos tapahtumapaikka löytyy listasta, osoitetta ja sijaintia ei tarvitse kuvailla tarkemmin. Voit kuitenkin laittaa lisätietoja tapahtuman löytämiseksi, kuten kerrosnumero tai muu tarkempi sijainti.",
      "editor-tip-location-not-found": "Jos tapahtumapaikkaa ei löydy listasta, valitse tapahtumapaikaksi Helsinki ja kirjoita tarkempi paikka tai osoite lisätietokenttään.",
      "editor-tip-price": "Merkitse onko tapahtuma maksuton. Syötä maksulliselle tapahtumalle hinta muodossa \\"5€\\".",
      "editor-tip-price-multi": "Jos tapahtumalla on useita hintoja, klikkaa Lisää uusi hintatieto ja merkitse kunkin hintaryhmän nimi (esim. Lapset) Hintatietojen kuvaus -kenttään.",
      "editor-tip-social-media": "Lisää linkki tapahtuman tai sen järjestäjän some-sivulle.",
      "editor-tip-time-delete": "Ylimääräisen ajankohdan voit poistaa valitsemalla ajankohdan vieressä olevan roskakorisymbolin.",
      "editor-tip-time-end": "Kirjoita myös päättymispäivä sekä päättymisaika, jos tapahtuma päättyy tiettyyn kellonaikaan.",
      "editor-tip-time-multi": "Jos tapahtuma järjestetään useamman kerran, voit lisätä tapahtumalle uusia ajankohtia. Jos tapahtuma toistuu säännöllisesti, voit lisätä kaikki ajankohdat kerralla valitsemalla Toistuva tapahtuma.",
      "editor-tip-time-start": "Kirjoita tapahtuman alkamispäivä ja myös alkamisaika, jos tapahtuma alkaa tiettyyn kellonaikaan.",
      "event-action-save-existing": "Tallenna muutokset julkaistuun tapahtumaan",
      "event-action-save-existing-active": "Tallennetaan muutoksia",
      "event-action-save-multiple": "Julkaise tapahtumat",
      "event-action-save-multiple-active": "Julkaistaan tapahtumia",
      "event-action-save-new": "Julkaise tapahtuma",
      "event-action-save-new-active": "Julkaistaan tapahtumaa",
      "event-add-new-occasion": "Lisää uusi ajankohta",
      "event-add-price": "Lisää uusi hintatieto",
      "event-add-recurring": "Toistuva tapahtuma",
      "event-categorization": "Tapahtuman luokittelu",
      "event-creation-cancel-success": "Tapahtuma peruttiin!",
      "event-creation-create-success": "Tapahtuma tallennettiin!",
      "event-creation-default-success": "Tapahtuma luotiin onnistuneesti!",
      "event-creation-delete-success": "Tapahtuma poistettiin!",
      "event-creation-multipleevents-success": "Tapahtumat tallennettiin!",
      "event-creation-publish-success": "Tapahtuma julkaistiin onnistuneesti!",
      "event-creation-savedraft-success": "Luonnoksen tallennus onnistui!",
      "event-creation-savepublic-success": "Julkaistun tapahtuman tallennus onnistui!",
      "event-creation-update-success": "Tapahtuma päivitettiin onnistuneesti!",
      "event-datetime-fields-header": "Tapahtuman ajankohdat",
      "event-description": "Kuvaus",
      "event-description-fields-header": "Tapahtuman kuvaus",
      "event-ending-datetime": "Tapahtuma päättyy",
      "event-headline": "Otsikko",
      "event-image": "Tapahtuman kuva (klikkaa vaihtaaksesi)",
      "event-info-url": "Kotisivu",
      "event-is-free": "Tapahtuma on maksuton",
      "event-keywords": "Tapahtuman asiasanat",
      "event-location": "Paikka",
      "event-location-additional-info": "Paikan lisätiedot",
      "event-location-fields-header": "Tapahtumapaikka",
      "event-location-id": "Paikan ID",
      "event-name-or-place": "Tapahtuman nimi tai paikka",
      "event-page-error": "Tapahtumaa ei löytynyt tai sinulla ei ole oikeuksia katsella sitä.",
      "event-page-loading": "Ladataan tapahtumaa...",
      "event-posted": "Tapahtuma lähetettiin onnistuneesti!",
      "event-presented-in-languages": "Tapahtumatiedot syötetään",
      "event-price": "Hinta",
      "event-price-fields-header": "Hintatiedot",
      "event-price-info": "Hintatietojen kuvaus",
      "event-provider": "Tapahtuman järjestäjä",
      "event-provider-input": "Tapahtuman järjestäjä (jos eri kuin syöttäjän organisaatio)",
      "event-purchase-link": "Linkki lipunmyyntiin/ilmoittautumiseen",
      "event-short-description": "Lyhyt kuvaus (korkeintaan 160 merkkiä)",
      "event-social-media-fields-header": "Tapahtuma sosiaalisessa mediassa",
      "event-sort-endtime": "Tapahtuma päättyy",
      "event-sort-last-modified": "Muokattu viimeksi",
      "event-sort-starttime": "Tapahtuma alkaa",
      "event-sort-title": "Otsikko",
      "event-starting-datetime": "Tapahtuma alkaa",
      "external-url-here": "Kuvan URL",
      "facebook-url": "Facebook (url)",
      "friday": "perjantai",
      "hel-event-languages": "Tapahtuman kielet",
      "hel-keywords": "Avainsanat",
      "hel-main-categories": "Hel.fi-pääkategoriat",
      "hel-target-groups": "Kohderyhmät",
      "image-caption-limit": "Kuvateksti (korkeintaan {limit} merkkiä)",
      "image-creation-error": "Virhe kuvaa luodessa.",
      "image-creation-success": "Kuva luotiin onnistuneesti!",
      "image-deletion-error": "Virhe kuvaa poistettaessa.",
      "image-deletion-success": "Kuva poistettiin onnistuneesti!",
      "image-update-error": "Virhe kuvaa päivittäessä.",
      "image-update-success": "Kuvan tiedot päivitettiin onnistuneesti!",
      "in-ar": "arabiaksi",
      "in-en": "englanniksi",
      "in-fi": "suomeksi",
      "in-ru": "venäjäksi",
      "in-sv": "ruotsiksi",
      "in-zh_hans": "kiinaksi",
      "instagram-url": "Instagram (url)",
      "is-free": "Maksuton",
      "lang-en": "Englanti",
      "lang-fi": "Suomi",
      "lang-sv": "Ruotsi",
      "login": "Kirjaudu sisään",
      "logout": "Kirjaudu ulos",
      "monday": "maanantai",
      "new-image": "Lisää uusi kuva",
      "no-connection": "Ei yhteyttä palvelimelle!",
      "no-image": "Tapahtumalla ei ole kuvaa",
      "no-validation-errors": "Lomake täytetty oikein!",
      "no-value": "-kenttää ei ole täytetty.",
      "or": "tai",
      "organization-events": "Tapahtumien hallinta",
      "organization-events-description": "Selaa ja muokkaa oman organisaatiosi julkisia tapahtumia sekä luonnoksia. Voit lisätä uuden tapahtuman yläpalkista.",
      "organization-events-no-results": "Yhtäkään muokattavaa tapahtumaa ei löytynyt.",
      "organization-events-prompt": "muokataksesi tapahtumia.",
      "organization-images": "Organisaation kuvat",
      "pick-time-range": "Etsi vain ajalta",
      "play-date-label": "Toistopäivät",
      "ready": "Valmis",
      "repeated": "Toistetaan ",
      "repetition-begin": "Toistuminen alkaa",
      "repetition-end": "Toistuminen päättyy",
      "repetition-end-time": "Tapahtumien loppuaika",
      "repetition-interval": " viikon välein.",
      "repetition-interval-label": "Toistoväli",
      "repetition-start-time": "Tapahtumien alkuaika",
      "saturday": "lauantai",
      "save": "Tallenna",
      "search-date-placeholder": "pp.kk.vvvv",
      "search-event-button": "Hae tapahtumia",
      "search-events": "Etsi tapahtumia",
      "search-events-description": "Katsele kaikkia Linked Eventsin tapahtumia.",
      "search-no-results": "Ei tuloksia",
      "server-error": "Virhe palvelimella.",
      "submit": "Lähetä",
      "sunday": "sunnuntai",
      "thursday": "torstai",
      "time": "Aika",
      "tuesday": "tiistai",
      "twitter-url": "Twitter (url)",
      "upload-image": "Lataa kuva kovalevyltäsi",
      "use-existing-image": "Käytä järjestelmässä olemassaolevaa kuvaa",
      "use-external-image-url": "tai syötä ulkoisen kuvan osoite:",
      "validation-afterStartTime": "Päättymisajan pitää olla alkamisaikaa myöhemmin",
      "validation-atLeastOne": "Valitse vähintään yksi kategoria",
      "validation-atLeastOneIsTrue": "Valitse vähintään yksi toistopäivä",
      "validation-daysWithinInterval": "Yksi tai useampi valituista päivistä ei esiinny toistovälillä",
      "validation-error": "Lomakkeessa on virheitä. Tarkasta lomake.",
      "validation-error-goto-error": "Siirry virheeseen",
      "validation-hasPrice": "Hintatietoja puuttuu",
      "validation-inTheFuture": "Tapahtuman pitää olla tulevaisuudessa",
      "validation-isDate": "Kirjoita päivämäärä muodossa pp.kk.vvvv tai p.k.vvvv",
      "validation-isMoreThanOne": "Toistovälin tulee olla vähintään 1 viikko",
      "validation-isTime": "Kirjoita aika oikeassa muodossa HH.mm tai HH",
      "validation-isUrl": "Kirjoita URL osoite kokonaisena ja oikeassa muodossa (http://...)",
      "validation-longStringLengthCounter": " merkkiä käytetty",
      "validation-offerIsFreeOrHasPrice": "Maksullinen tapahtuma vaatii hinnan kaikilla valituilla kielillä",
      "validation-requiredAtId": "Tämä kenttä vaaditaan",
      "validation-requiredInContentLanguages": "Tämä kenttä vaaditaan kaikilla valituilla kielillä",
      "validation-requiredMulti": "Tämä kenttä vaaditaan",
      "validation-requiredString": "Tämä kenttä vaaditaan",
      "validation-stringLengthCounter": "{counter} merkkiä jäljellä",
      "validation-stringLimitReached": "Tämä kenttä voi olla korkeintaan {limit} merkkiä pitkä",
      "wednesday": "keskiviikko",
      "welcome": "Tervetuloa",
    }
  }
>
  <MemoryRouter>
    <Connect(InjectIntl(EditorPage))
      match={
        Object {
          "params": Object {
            "action": "update",
            "eventId": "helsinki:afqxukccli",
          },
        }
      }
    />
  </MemoryRouter>
</IntlProvider>
`;

exports[`Editor Snapshot should render view correctly when new event 1`] = `
<IntlProvider
  locale="fi"
  messages={
    Object {
      "add-more": "Lisää kerrat",
      "authorization-required": "Kirjaudu sisään tapahtuman lähettämiseksi.",
      "cancel": "Kumoa",
      "cancel-event": "Peruuta tapahtuma",
      "cancelled": "Peruutettu",
      "choose-image": "Lisää tapahtumalle kuva tai valitse kuvapankista",
      "clear-form": "Tyhjennä kentät",
      "confirm": "Suorita",
      "confirm-action-header": "Varmista toiminto",
      "confirm-cancel": "Varmista tapahtuman peruminen",
      "confirm-delete": "Varmista tapahtuman poistaminen",
      "confirm-image-delete": "Varmista kuvan poistaminen",
      "copy-event-to-draft": "Kopioi uuden tapahtuman pohjaksi",
      "copy-to-clipboard": "Kopioi leikepöydälle",
      "create-event": "Lisää uusi tapahtuma",
      "date": "Päivämäärä",
      "delete": "Poista",
      "delete-event": "Poista tapahtuma",
      "draft": "Luonnos",
      "edit-event": "Muokkaa tapahtumaa",
      "editor-delete-subevents-warning": "Poistaessasi tämän tapahtuman myös seuraavat alitapahtumat poistetaan:",
      "editor-delete-warning": "VAROITUS: Tämä toiminto poistaa tapahtuman lopullisesti. Voit tarvittaessa myös perua tapahtuman tai lykätä sitä.",
      "editor-sentinel-alert": "Voit katsella lomaketta, mutta sinulla ei ole oikeuksia julkaista tai muokata tapahtumia. Et ole kirjautunut sisään tai kirjautumisesi on vanhentunut.",
      "editor-tip-event-languages": "Kielet, joita tapahtumassa käytetään.",
      "editor-tip-hel-main-category": "Valitse vähintään yksi pääkategoria.",
      "editor-tip-hel-target-group": "Jos tapahtumalla ei ole erityistä kohderyhmää, älä valitse mitään.",
      "editor-tip-keywords": "Liitä tapahtumaan vähintään yksi asiasana, joka kuvaa tapahtuman teemaa. Aloita kirjoittamaan asiasanaa ja valitse lisättävä asiasana alle ilmestyvästä listasta.",
      "editor-tip-location": "Aloita kirjoittamaan kenttään tapahtumapaikan nimen alkua ja valitse oikea paikka alle ilmestyvästä listasta.",
      "editor-tip-location-extra": "Jos tapahtumapaikka löytyy listasta, osoitetta ja sijaintia ei tarvitse kuvailla tarkemmin. Voit kuitenkin laittaa lisätietoja tapahtuman löytämiseksi, kuten kerrosnumero tai muu tarkempi sijainti.",
      "editor-tip-location-not-found": "Jos tapahtumapaikkaa ei löydy listasta, valitse tapahtumapaikaksi Helsinki ja kirjoita tarkempi paikka tai osoite lisätietokenttään.",
      "editor-tip-price": "Merkitse onko tapahtuma maksuton. Syötä maksulliselle tapahtumalle hinta muodossa \\"5€\\".",
      "editor-tip-price-multi": "Jos tapahtumalla on useita hintoja, klikkaa Lisää uusi hintatieto ja merkitse kunkin hintaryhmän nimi (esim. Lapset) Hintatietojen kuvaus -kenttään.",
      "editor-tip-social-media": "Lisää linkki tapahtuman tai sen järjestäjän some-sivulle.",
      "editor-tip-time-delete": "Ylimääräisen ajankohdan voit poistaa valitsemalla ajankohdan vieressä olevan roskakorisymbolin.",
      "editor-tip-time-end": "Kirjoita myös päättymispäivä sekä päättymisaika, jos tapahtuma päättyy tiettyyn kellonaikaan.",
      "editor-tip-time-multi": "Jos tapahtuma järjestetään useamman kerran, voit lisätä tapahtumalle uusia ajankohtia. Jos tapahtuma toistuu säännöllisesti, voit lisätä kaikki ajankohdat kerralla valitsemalla Toistuva tapahtuma.",
      "editor-tip-time-start": "Kirjoita tapahtuman alkamispäivä ja myös alkamisaika, jos tapahtuma alkaa tiettyyn kellonaikaan.",
      "event-action-save-existing": "Tallenna muutokset julkaistuun tapahtumaan",
      "event-action-save-existing-active": "Tallennetaan muutoksia",
      "event-action-save-multiple": "Julkaise tapahtumat",
      "event-action-save-multiple-active": "Julkaistaan tapahtumia",
      "event-action-save-new": "Julkaise tapahtuma",
      "event-action-save-new-active": "Julkaistaan tapahtumaa",
      "event-add-new-occasion": "Lisää uusi ajankohta",
      "event-add-price": "Lisää uusi hintatieto",
      "event-add-recurring": "Toistuva tapahtuma",
      "event-categorization": "Tapahtuman luokittelu",
      "event-creation-cancel-success": "Tapahtuma peruttiin!",
      "event-creation-create-success": "Tapahtuma tallennettiin!",
      "event-creation-default-success": "Tapahtuma luotiin onnistuneesti!",
      "event-creation-delete-success": "Tapahtuma poistettiin!",
      "event-creation-multipleevents-success": "Tapahtumat tallennettiin!",
      "event-creation-publish-success": "Tapahtuma julkaistiin onnistuneesti!",
      "event-creation-savedraft-success": "Luonnoksen tallennus onnistui!",
      "event-creation-savepublic-success": "Julkaistun tapahtuman tallennus onnistui!",
      "event-creation-update-success": "Tapahtuma päivitettiin onnistuneesti!",
      "event-datetime-fields-header": "Tapahtuman ajankohdat",
      "event-description": "Kuvaus",
      "event-description-fields-header": "Tapahtuman kuvaus",
      "event-ending-datetime": "Tapahtuma päättyy",
      "event-headline": "Otsikko",
      "event-image": "Tapahtuman kuva (klikkaa vaihtaaksesi)",
      "event-info-url": "Kotisivu",
      "event-is-free": "Tapahtuma on maksuton",
      "event-keywords": "Tapahtuman asiasanat",
      "event-location": "Paikka",
      "event-location-additional-info": "Paikan lisätiedot",
      "event-location-fields-header": "Tapahtumapaikka",
      "event-location-id": "Paikan ID",
      "event-name-or-place": "Tapahtuman nimi tai paikka",
      "event-page-error": "Tapahtumaa ei löytynyt tai sinulla ei ole oikeuksia katsella sitä.",
      "event-page-loading": "Ladataan tapahtumaa...",
      "event-posted": "Tapahtuma lähetettiin onnistuneesti!",
      "event-presented-in-languages": "Tapahtumatiedot syötetään",
      "event-price": "Hinta",
      "event-price-fields-header": "Hintatiedot",
      "event-price-info": "Hintatietojen kuvaus",
      "event-provider": "Tapahtuman järjestäjä",
      "event-provider-input": "Tapahtuman järjestäjä (jos eri kuin syöttäjän organisaatio)",
      "event-purchase-link": "Linkki lipunmyyntiin/ilmoittautumiseen",
      "event-short-description": "Lyhyt kuvaus (korkeintaan 160 merkkiä)",
      "event-social-media-fields-header": "Tapahtuma sosiaalisessa mediassa",
      "event-sort-endtime": "Tapahtuma päättyy",
      "event-sort-last-modified": "Muokattu viimeksi",
      "event-sort-starttime": "Tapahtuma alkaa",
      "event-sort-title": "Otsikko",
      "event-starting-datetime": "Tapahtuma alkaa",
      "external-url-here": "Kuvan URL",
      "facebook-url": "Facebook (url)",
      "friday": "perjantai",
      "hel-event-languages": "Tapahtuman kielet",
      "hel-keywords": "Avainsanat",
      "hel-main-categories": "Hel.fi-pääkategoriat",
      "hel-target-groups": "Kohderyhmät",
      "image-caption-limit": "Kuvateksti (korkeintaan {limit} merkkiä)",
      "image-creation-error": "Virhe kuvaa luodessa.",
      "image-creation-success": "Kuva luotiin onnistuneesti!",
      "image-deletion-error": "Virhe kuvaa poistettaessa.",
      "image-deletion-success": "Kuva poistettiin onnistuneesti!",
      "image-update-error": "Virhe kuvaa päivittäessä.",
      "image-update-success": "Kuvan tiedot päivitettiin onnistuneesti!",
      "in-ar": "arabiaksi",
      "in-en": "englanniksi",
      "in-fi": "suomeksi",
      "in-ru": "venäjäksi",
      "in-sv": "ruotsiksi",
      "in-zh_hans": "kiinaksi",
      "instagram-url": "Instagram (url)",
      "is-free": "Maksuton",
      "lang-en": "Englanti",
      "lang-fi": "Suomi",
      "lang-sv": "Ruotsi",
      "login": "Kirjaudu sisään",
      "logout": "Kirjaudu ulos",
      "monday": "maanantai",
      "new-image": "Lisää uusi kuva",
      "no-connection": "Ei yhteyttä palvelimelle!",
      "no-image": "Tapahtumalla ei ole kuvaa",
      "no-validation-errors": "Lomake täytetty oikein!",
      "no-value": "-kenttää ei ole täytetty.",
      "or": "tai",
      "organization-events": "Tapahtumien hallinta",
      "organization-events-description": "Selaa ja muokkaa oman organisaatiosi julkisia tapahtumia sekä luonnoksia. Voit lisätä uuden tapahtuman yläpalkista.",
      "organization-events-no-results": "Yhtäkään muokattavaa tapahtumaa ei löytynyt.",
      "organization-events-prompt": "muokataksesi tapahtumia.",
      "organization-images": "Organisaation kuvat",
      "pick-time-range": "Etsi vain ajalta",
      "play-date-label": "Toistopäivät",
      "ready": "Valmis",
      "repeated": "Toistetaan ",
      "repetition-begin": "Toistuminen alkaa",
      "repetition-end": "Toistuminen päättyy",
      "repetition-end-time": "Tapahtumien loppuaika",
      "repetition-interval": " viikon välein.",
      "repetition-interval-label": "Toistoväli",
      "repetition-start-time": "Tapahtumien alkuaika",
      "saturday": "lauantai",
      "save": "Tallenna",
      "search-date-placeholder": "pp.kk.vvvv",
      "search-event-button": "Hae tapahtumia",
      "search-events": "Etsi tapahtumia",
      "search-events-description": "Katsele kaikkia Linked Eventsin tapahtumia.",
      "search-no-results": "Ei tuloksia",
      "server-error": "Virhe palvelimella.",
      "submit": "Lähetä",
      "sunday": "sunnuntai",
      "thursday": "torstai",
      "time": "Aika",
      "tuesday": "tiistai",
      "twitter-url": "Twitter (url)",
      "upload-image": "Lataa kuva kovalevyltäsi",
      "use-existing-image": "Käytä järjestelmässä olemassaolevaa kuvaa",
      "use-external-image-url": "tai syötä ulkoisen kuvan osoite:",
      "validation-afterStartTime": "Päättymisajan pitää olla alkamisaikaa myöhemmin",
      "validation-atLeastOne": "Valitse vähintään yksi kategoria",
      "validation-atLeastOneIsTrue": "Valitse vähintään yksi toistopäivä",
      "validation-daysWithinInterval": "Yksi tai useampi valituista päivistä ei esiinny toistovälillä",
      "validation-error": "Lomakkeessa on virheitä. Tarkasta lomake.",
      "validation-error-goto-error": "Siirry virheeseen",
      "validation-hasPrice": "Hintatietoja puuttuu",
      "validation-inTheFuture": "Tapahtuman pitää olla tulevaisuudessa",
      "validation-isDate": "Kirjoita päivämäärä muodossa pp.kk.vvvv tai p.k.vvvv",
      "validation-isMoreThanOne": "Toistovälin tulee olla vähintään 1 viikko",
      "validation-isTime": "Kirjoita aika oikeassa muodossa HH.mm tai HH",
      "validation-isUrl": "Kirjoita URL osoite kokonaisena ja oikeassa muodossa (http://...)",
      "validation-longStringLengthCounter": " merkkiä käytetty",
      "validation-offerIsFreeOrHasPrice": "Maksullinen tapahtuma vaatii hinnan kaikilla valituilla kielillä",
      "validation-requiredAtId": "Tämä kenttä vaaditaan",
      "validation-requiredInContentLanguages": "Tämä kenttä vaaditaan kaikilla valituilla kielillä",
      "validation-requiredMulti": "Tämä kenttä vaaditaan",
      "validation-requiredString": "Tämä kenttä vaaditaan",
      "validation-stringLengthCounter": "{counter} merkkiä jäljellä",
      "validation-stringLimitReached": "Tämä kenttä voi olla korkeintaan {limit} merkkiä pitkä",
      "wednesday": "keskiviikko",
      "welcome": "Tervetuloa",
    }
  }
>
  <MemoryRouter>
    <Connect(InjectIntl(EditorPage))
      match={
        Object {
          "params": Object {
            "action": "create",
            "eventId": "new?_k=dn954b",
          },
        }
      }
    />
  </MemoryRouter>
</IntlProvider>
`;
